app/parse.py

```diff
@@ -15,9 +37,81 @@ class Product:
     num_of_reviews: int
 
 
+_driver: WebDriver | None = None
+
+
+def get_driver() -> WebDriver:
+    return _driver
+
+
+def set_driver(new_driver: WebDriver) -> None:
+    global _driver
+    _driver = new_driver
+
+
+def parse_single_product(product_soup: BeautifulSoup) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(".description").text,
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=int(len(product_soup.select(".glyphicon"))),
+        num_of_reviews=int(
+            product_soup.select_one(".ratings > p").text.split()[0]
+        ),
+    )
+
+
+def get_page_soup(url: str) -> list[Product]:
+    page = requests.get(url).content
+    page_soup = BeautifulSoup(page, "html.parser")
+
+    if page_soup.select_one(".ecomerce-items-scroll-more"):
+        with webdriver.Chrome() as new_driver:
+            set_driver(new_driver)
+            driver = get_driver()
+            driver.get(url)
+            button = driver.find_element(
+                By.CLASS_NAME, "ecomerce-items-scroll-more"
+            )
+            driver.find_element(By.CLASS_NAME, "acceptCookies").click()
+
+            while not button.get_property("style"):
+                button.click()
+                time.sleep(0.1)
+
+                button = driver.find_element(
+                    By.CLASS_NAME, "ecomerce-items-scroll-more"
+                )
+
+            with open("index.html", "w", encoding="cp1251") as file:
+                file.write(driver.page_source)
+
+        with open("index.html") as file:
+            page = file.read().replace("&nbsp;", " ")
+        os.remove("index.html")
```

Comment:
Can't you parse driver.page_source directly without creating an additional file?

Reply:
I can, it's just harder debug, but I will redo it now))